XenAdmin/MainWindow.cs

```diff
@@ -972,6 +972,34 @@ void connection_CachePopulated(IXenConnection connection)
                     });
                     return;
                 }
+
+                // Allow Citrix Hypervisor Center connect to Havana, Stockholm and cloud released CH only
+                //
+                if ((int)API_Version.LATEST >= (int)API_Version.API_2_16

```

Comment:
`(int)API_Version.LATEST >= (int)API_Version.API_2_16` is not needed, as this will always be true

For checking the Havana version, we could use the existing `HavanaOrGreater` function which returns true for 7.1 CU2 (Havana) and any other future CUs we might have on 7.1

Reply:
Regarding first comment, Tina is intending to share same source code for XC 8.2.x and hyperman, and doing some manual tests to assess the feasibility.
Second comment fixed.

Reply:
This check now allows anything from havana to stockholm. I think we need this:
```
current_version >= (int)API_Version.API_2_16 &&
server_min < (int)API_Version.API_2_15 &&
(server_min != (int)API_Version.API_2_6 || !Helpers.HavanaOrGreater(master))
```

Reply:
I think HavanaOrGreater() only hit 7.1 CU2 at this moment instead of all versions from Havana to Stockholm, because "!FalconOrGreater()" is included in HavanaOrGreater() function.

Reply:
Ah true, I hadn't looked thoroughly into HavanaOrGreater() (perhaps it should probably be called something else because what it actually does is different from what it says). Ok, ignore.

Reply:
Yes, the name is confusing and the function is different from our other XOrGreater functions, but in this case it's exactly what we need